publish-to-github/action.yml

```
@@ -26,7 +26,7 @@ runs:
 
   steps:
     - name: Download all the dists
-      uses: actions/download-artifact@v3
+      uses: actions/download-artifact@v4
       with:
         name: python-package-distributions
         path: dist/
```

Assuming we should change to this

```suggestion
        path: dist-${{ matrix.runs-on }}/
```